spec/requests/login/authentication_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'POST /login', type: :request do
+  let(:user) { create(:user) }
+  let(:url) { '/login' }
+  let(:params) do
+    {
+      user: {
+        email: user.email,
+        password: user.password
+      }
+    }
+  end
+
+  context 'when params are correct' do
+    before do
+      post url, params: params
+    end
+
+    it 'returns 200' do
+      expect(response).to have_http_status(200)
+    end
+
+    it 'returns JTW token in authorization header' do
+      expect(response.headers['Authorization']).to be_present
+    end
+
+    it 'returns valid JWT token' do
+      token_from_request = response.headers['Authorization'].split(' ').last
+      decoded_token = JWT.decode(token_from_request, ENV['DEVISE_JWT_SECRET_KEY'], true)
+      expect(decoded_token.first['sub']).to be_present
+    end
+  end
+
+  context 'when login params are incorrect' do
+    before { post url }
+
+    it 'returns unathorized status' do
```

Comment:
This can be refactored [https://www.youtube.com/watch?v=f0I6a0AntP0](https://www.youtube.com/watch?v=f0I6a0AntP0)

Reply:
create a refactoring ticket for this.